src/Version10/Definition/Builder/ConsoleOptionBuilderInterface.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phpcq\PluginApi\Version10\Definition\Builder;
+
+/**
+ * This interface describes a command option.
+ */
+interface ConsoleOptionBuilderInterface
+{
+    public const VALUE_SEPARATOR_SPACE = ' ';
+
+    public const VALUE_SEPARATOR_EQUAL_SIGN = '=';
+
+    public const VALUE_SEPARATOR_NONE = '';
+
+    /**
+     * Mark the option as required. If not defined the option is considered as optional.
+     *
+     * @return self
+     */
+    public function isRequired(): self;
+
+    /**
+     * Mark the option as an array if the option may occur multiple times.
+     *
+     * @return self
+     */
+    public function isArray(): self;
+
+    /**
+     * Declare that a value is required for this option.
+     *
+     * @return self
+     */
+    public function withRequiredValue(): self;
```

Comment:
Is mutually exclusive with `ConsoleOptionBuilderInterface::withOptionalValue()` I suppose?

See also remarks below for `ConsoleOptionBuilderInterface::withOptionalValue()`